### PR TITLE
Configure message list

### PR DESCRIPTION
**Pull request by feodal01**
1) New method "show_context" = show all context now in use
2) Now messages filtering based on token length in messages, not by count of messages. It should fix problem with "please reset context, it is too long"

### CODE REVIEW

File: src/db_utils/scheme.py [hunk 1]
```diff
@@ -82,6 +96,57 @@ def set_current_context(user: User, context_name: str) -> Optional[int]:
...
+        filtered_messages.append({'role': 'system', 'content': system_prompt})
+        massage_length += num_tokens_from_string(system_prompt)
+
+    # make last user message available as last message
+    last_message = messages_copy[-1]['content']
+    massage_length += num_tokens_from_string(last_message)
+    filtered_messages.append(parse_messages(messages_copy[-1]))
+    messages_copy.pop(-1)
+
+    if len(messages_copy) != 0:
```
Comment (mrYush) [inline src/db_utils/scheme.py]: extra conditions

File: src/db_utils/scheme.py [hunk 2]
```diff
@@ -82,6 +96,57 @@ def set_current_context(user: User, context_name: str) -> Optional[int]:
...
+                              system_prompt: str = None):
+    all_messages = ConversationCollection.objects(telegram_id=user_id)
+    all_messages_list = [{'role': msg.role, 'content': msg.content, 'timestamp': msg.timestamp}
+                         for msg in all_messages]
+
+    if len(all_messages_list) == 0:
+        return list()
+
+    all_messages_list_sorted = sorted(all_messages_list, key=lambda d: d['timestamp'])
+    messages_copy = deepcopy(all_messages_list_sorted)
```
Comment (mrYush) [inline src/db_utils/scheme.py]: not necessary, we can use all_messages_list

File: src/db_utils/scheme.py [hunk 3]
```diff
@@ -82,6 +96,57 @@ def set_current_context(user: User, context_name: str) -> Optional[int]:
...
+
+
+def get_last_n_message_tokens(user_id: int,
+                              tokens: int = MAX_TOKENS_CONTEXT_HISTORY,
+                              system_prompt: str = None):
+    all_messages = ConversationCollection.objects(telegram_id=user_id)
+    all_messages_list = [{'role': msg.role, 'content': msg.content, 'timestamp': msg.timestamp}
+                         for msg in all_messages]
+
+    if len(all_messages_list) == 0:
```
Comment (mrYush) [inline src/db_utils/scheme.py]: imposssible... must rase at least warning... and handle this case in bot

File: src/main.py [hunk 4]
```diff
@@ -42,6 +43,9 @@ def main() -> None:
     application.add_handler(
         CommandHandler("context", choose_context)
     )
+    application.add_handler(
+        CommandHandler('show_context', show_context)
```
Comment (mrYush) [inline src/main.py]: SHOW_CONTEXT

File: src/db_utils/scheme.py (hunk 1 above)
Comment (feodal01) [reply src/db_utils/scheme.py]: del

File: src/db_utils/scheme.py (hunk 3 above)
Comment (feodal01) [reply src/db_utils/scheme.py]: del as there us check in next steps

**Review by mrYush — APPROVED** [review]

**Pull request merged**